package kotlin;
/* loaded from: classes6.dex */
public final /* synthetic */ class e {
    public static /* synthetic */ long a(long j4, long j5) {
        if (j5 < 0) {
            return (j4 ^ Long.MIN_VALUE) < (Long.MIN_VALUE ^ j5) ? j4 : j4 - j5;
        } else if (j4 >= 0) {
            return j4 % j5;
        } else {
            long j6 = j4 - ((((j4 >>> 1) / j5) << 1) * j5);
            if ((j6 ^ Long.MIN_VALUE) < (Long.MIN_VALUE ^ j5)) {
                j5 = 0;
            }
            return j6 - j5;
        }
    }
}
